prac_05/state_names.py

```diff
@@ -0,0 +1,25 @@
+"""
+CP1404/CP5632 Practical
+State names in a dictionary
+File needs reformatting
+"""
+
+CODE_TO_NAME = {"QLD": "Queensland", "NSW": "New South Wales", "NT": "Northern Territory", "WA": "Western Australia",
+                "ACT": "Australian Capital Territory", "VIC": "Victoria", "TAS": "Tasmania"}
+print(CODE_TO_NAME)
+
+# EAFP approach
+state_code = input("Enter short state: ").upper()
+while state_code != "":
+    try:
+        print(state_code, "is", CODE_TO_NAME[state_code])
+    except KeyError:
+        print("Invalid short state")
+    state_code = input("Enter short state: ").upper()
+
+for state, state_name in CODE_TO_NAME.items():
+    max_state_length = max(len(state) for state in CODE_TO_NAME.keys())
```

Comment:
These list comprehension functions will go through all the items in CODE_TO_NAME and find the longest state and state name. Do they have to be in a for loop then...